### PR TITLE
filters/auth: add setRequestHeaderFromSecret filter

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Add a filter to set request header value from secret with optional prefix and suffix.

It is similar to `bearerinjector` which is equivalent to `setRequestHeaderFromSecret("Authorization", "/tokens/my-token", "Bearer ")`

For #1952

### CODE REVIEW

File: filters/auth/secretheader_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package auth_test
```
Comment (AlexanderYastrebov) [inline filters/auth/secretheader_test.go:1]: Tests cover 100% of the code

File: docs/reference/filters.md [hunk 2]
```diff
@@ -2957,31 +2957,54 @@ the -rfc-patch-path flag. See
...
-`/tmp/secrets/write-token` and `/tmp/secrets/read-token`.
+`/tmp/secrets/get-token`, `/tmp/secrets/post-token` and `/tmp/secrets/put-token`.
+
+### bearerinjector
+
+This filter injects `Bearer` tokens into `Authorization` headers read
+from file providing the token as content.
+
+It is a special form of `setRequestHeaderFromSecret` with `"Authorization"` header name,
+`"Bearer "` prefix and empty suffix.
```
Comment (szuecs) [inline docs/reference/filters.md:2999]: If so we could replace the implementation of bearerinjector by a configuration of setRequestHeaderFromSecret, so less code and better tested.
Wdyt?

**Comment by szuecs** [comment]
:+1:

File: docs/reference/filters.md (hunk 2 above)
Comment (AlexanderYastrebov) [reply docs/reference/filters.md:2999]: I thought about it but then we would need to limit number of arguments as bearerinjector only requires one so I decided ti does not worth it.

Comment (szuecs) [reply docs/reference/filters.md:2999]: ok

**Comment by MustafaSaber** [comment]
👍

**Pull request merged**